### PR TITLE
Fixes #1, resolves intermittent timing issue 

### LINKED ISSUE #1: MathJax not loading quick enough from CDN when previewing
When using the MathJax plugin, and previewing locally the script is not retrieved form the CDN quick enough. This leads to MathJax content not being processed on the page.

### PR DESCRIPTION
**Pull request by brian-learningpool**
- Added new loadScript() function which adds the config, init and CDN script
- This has a callback which waits until one script is loaded before processing the next
- Added a polling function on onViewReady() to check that Hub actually exists before trying to call Queue()

### CODE REVIEW

File: js/adapt-mathJax.js [hunk 1]
```diff
@@ -27,11 +59,18 @@ define([ "core/js/adapt" ], function(Adapt) {
...
 
 	function onViewReady(view) {
-		var Hub = window.MathJax.Hub;
-
 		$(".loading").show();
 
-		if (Hub) Hub.Queue([ "Typeset", Hub, view.el ]);
+		function checkForMathJax() {
+			if (!window.MathJax || !window.MathJax.Hub) {
+				window.setTimeout(checkForLibrary, 200);
```
Comment (oliverfoster) [inline js/adapt-mathJax.js]: checkForLibrary > checkForMathJax ?

Comment (brian-learningpool) [reply js/adapt-mathJax.js]: Nice catch.

**Comment by simondate** [comment]
I've found the current master branch seems to have some issues when you try to load the file over `https`, `http` worked fine funnily enough.

Brian's PR has made it work correctly all the time.

**Comment by brian-learningpool** [comment]
I'd actually forgotten all about this PR.  Is there any chance of getting this merged?

@simondate, I've just merged the latest `master` branch in.  If you get a moment can you verify it still works ok?

**Comment by simondate** [comment]
@brian-learningpool `2.7.2` works great for me!

**Comment by moloko** [comment]
if @simondate is saying it works and @oliverfoster is happy his changes have been done then I'm OK to merge.

File: js/adapt-mathJax.js [hunk 2]
```diff
@@ -1,20 +1,50 @@
...
+			script.onreadystatechange = callback;
+			script.onload = callback;
+		}
+    
+        // Append the <script> tag.
+        head.appendChild(script);
+	}
+
 	function setUpMathJax() {
+		Adapt.trigger('plugin:beginWait');
```
Comment (moloko) [inline js/adapt-mathJax.js]: does this create a dependency on Framework [v2.0.16](https://github.com/adaptlearning/adapt_framework/releases/tag/v2.0.16) or would it still work OK if you had an earlier version?

File: js/adapt-mathJax.js [hunk 3]
```diff
@@ -1,20 +1,50 @@
...
-			"//cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js";
-		var $init = $("<script/>", { src: "assets/mathJaxInit.js" });
-		var $config = $("<script/>", { type: "text/x-mathjax-config" });
-		var $src = $("<script/>", { src: src });
-
-		$config[0].text = "MathJax.Hub.Config(" + JSON.stringify(inlineConfig) + ");";
-		$("head").append($init, $config, $src);
+		var src = config ? config._src : "//cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js";
+
+		loadScript({
```
Comment (moloko) [inline js/adapt-mathJax.js:38]: sorry if this is a bit of a daft question but could `$.getScript()` not handle this? If not, no need to explain why - just say 'no'!

Comment (brian-learningpool) [reply js/adapt-mathJax.js:38]: No.  Although the function is called `loadScript()` it takes handles both URLs and arbitrary script and is not really equivalent.

File: js/adapt-mathJax.js (hunk 2 above)
Comment (brian-learningpool) [reply js/adapt-mathJax.js]: It still works, though possibly intermittently on an earlier version.  I'll add a patch to correct the deprecation warning in v3.

**Review by tomgreenfield — APPROVED** [review]
A blind test from me (cannot replicate original issue) but happy to +1 if works for everyone.

**Pull request merged**